### PR TITLE
Add type annotations to IndexOpsMixin(base.py)

### PR DESCRIPTION
**Pull request by xinrong-databricks**

### CODE REVIEW

File: databricks/koalas/base.py [hunk 1]
```diff
@@ -178,7 +178,7 @@ def spark_column(self) -> Column:
     # arithmetic operators
     __neg__ = column_op(Column.__neg__)
 
-    def __add__(self, other):
+    def __add__(self, other) -> "ks.Series":
```
Comment (ueshin) [inline databricks/koalas/base.py:181]: `Union["ks.Series", "ks.Index"]`? and ditto for the following binary operations?

File: databricks/koalas/base.py [hunk 2]
```diff
@@ -558,7 +558,7 @@ def is_monotonic(self):
 
         Returns
         -------
-        is_monotonic : boolean
+        is_monotonic : bool
```
Comment (ueshin) [inline databricks/koalas/base.py:561]: Shall we fix it in `is_monotonic_decreasing` as well?

File: databricks/koalas/base.py [hunk 3]
```diff
@@ -839,7 +839,7 @@ def astype(self, dtype):
             raise ValueError("Type {} not understood".format(dtype))
         return self._with_new_scol(self.spark.column.cast(spark_type))
 
-    def isin(self, values):
+    def isin(self, values) -> "ks.Series":
```
Comment (ueshin) [inline databricks/koalas/base.py:842]: `Union["ks.Series", "ks.Index"]`?

File: databricks/koalas/base.py [hunk 4]
```diff
@@ -891,7 +891,7 @@ def isin(self, values):
 
         return self._with_new_scol(self.spark.column.isin(list(values)))
 
-    def isnull(self):
+    def isnull(self) -> "ks.Series":
```
Comment (ueshin) [inline databricks/koalas/base.py:894]: ditto.

File: databricks/koalas/base.py [hunk 5]
```diff
@@ -929,7 +929,7 @@ def isnull(self):
 
     isna = isnull
 
-    def notnull(self):
+    def notnull(self) -> "ks.Series":
```
Comment (ueshin) [inline databricks/koalas/base.py:932]: ditto.

File: databricks/koalas/base.py (hunk 3 above)
Comment (xinrong-databricks) [reply databricks/koalas/base.py]: Sounds good! Updated the document as well!

File: databricks/koalas/base.py (hunk 2 above)
Comment (xinrong-databricks) [reply databricks/koalas/base.py:561]: Fixed.

File: databricks/koalas/base.py (hunk 1 above)
Comment (xinrong-databricks) [reply databricks/koalas/base.py]: Makes sense! Modified.

File: databricks/koalas/base.py (hunk 3 above)
Comment (ueshin) [reply databricks/koalas/base.py]: Nice!

**Review by ueshin — APPROVED** [review]

**Comment by ueshin** [comment]
Thanks! merging.

**Pull request merged**